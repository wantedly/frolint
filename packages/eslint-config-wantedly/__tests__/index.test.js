const ESLint = require("eslint").ESLint;
const { base, react } = require("../index");

describe("eslint-config-wantedly", () => {
  describe("base", () => {
    test("should match snapshot for", async () => {
      const config = await new ESLint({
        baseConfig: base,
        overrideConfigFile: true,
      }).calculateConfigForFile("test.js");
      expect(config).toMatchSnapshot();
    });
  });

  describe("react", () => {
    test("should match snapshot for", async () => {
      const config = await new ESLint({
        baseConfig: react,
        overrideConfigFile: true,
      }).calculateConfigForFile("test.js");
      expect(config).toMatchSnapshot();
    });
  });
});
