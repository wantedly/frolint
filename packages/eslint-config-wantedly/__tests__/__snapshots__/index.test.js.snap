// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-config-wantedly should match snapshot for: env 1`] = `
Object {
  "browser": true,
  "es6": true,
  "jest/globals": true,
  "node": true,
}
`;

exports[`eslint-config-wantedly should match snapshot for: globals 1`] = `
Object {
  "flushPromises": true,
}
`;

exports[`eslint-config-wantedly should match snapshot for: ignorePatterns 1`] = `Array []`;

exports[`eslint-config-wantedly should match snapshot for: noInlineConfig 1`] = `undefined`;

exports[`eslint-config-wantedly should match snapshot for: parser 1`] = `"/babel-eslint/lib/index.js"`;

exports[`eslint-config-wantedly should match snapshot for: parserOptions 1`] = `
Object {
  "ecmaFeatures": Object {
    "experimentalObjectRestSpread": true,
    "jsx": true,
  },
  "sourceType": "module",
}
`;

exports[`eslint-config-wantedly should match snapshot for: plugins 1`] = `
Array [
  "react-hooks",
  "react",
  "use-macros",
  "prettier",
  "jest",
  "jsx-a11y",
  "import",
]
`;

exports[`eslint-config-wantedly should match snapshot for: reportUnusedDisableDirectives 1`] = `undefined`;

exports[`eslint-config-wantedly should match snapshot for: rules 1`] = `
Object {
  "@typescript-eslint/brace-style": Array [
    "off",
  ],
  "@typescript-eslint/comma-dangle": Array [
    "off",
  ],
  "@typescript-eslint/comma-spacing": Array [
    "off",
  ],
  "@typescript-eslint/func-call-spacing": Array [
    "off",
  ],
  "@typescript-eslint/indent": Array [
    "off",
  ],
  "@typescript-eslint/keyword-spacing": Array [
    "off",
  ],
  "@typescript-eslint/member-delimiter-style": Array [
    "off",
  ],
  "@typescript-eslint/no-extra-parens": Array [
    "off",
  ],
  "@typescript-eslint/no-extra-semi": Array [
    "off",
  ],
  "@typescript-eslint/object-curly-spacing": Array [
    "off",
  ],
  "@typescript-eslint/quotes": Array [
    0,
  ],
  "@typescript-eslint/semi": Array [
    "off",
  ],
  "@typescript-eslint/space-before-function-paren": Array [
    "off",
  ],
  "@typescript-eslint/space-infix-ops": Array [
    "off",
  ],
  "@typescript-eslint/type-annotation-spacing": Array [
    "off",
  ],
  "array-bracket-newline": Array [
    "off",
  ],
  "array-bracket-spacing": Array [
    "off",
  ],
  "array-callback-return": Array [
    "off",
  ],
  "array-element-newline": Array [
    "off",
  ],
  "arrow-body-style": Array [
    "off",
  ],
  "arrow-parens": Array [
    "warn",
    "always",
  ],
  "arrow-spacing": Array [
    "off",
  ],
  "babel/object-curly-spacing": Array [
    "off",
  ],
  "babel/quotes": Array [
    0,
  ],
  "babel/semi": Array [
    "off",
  ],
  "block-spacing": Array [
    "off",
  ],
  "brace-style": Array [
    "off",
  ],
  "camelcase": Array [
    "error",
    Object {
      "ignoreDestructuring": false,
      "ignoreGlobals": false,
      "ignoreImports": false,
      "properties": "never",
    },
  ],
  "class-methods-use-this": Array [
    "off",
  ],
  "comma-dangle": Array [
    "error",
    Object {
      "arrays": "always-multiline",
      "exports": "always-multiline",
      "functions": "never",
      "imports": "always-multiline",
      "objects": "always-multiline",
    },
  ],
  "comma-spacing": Array [
    "off",
  ],
  "comma-style": Array [
    "off",
  ],
  "computed-property-spacing": Array [
    "off",
  ],
  "consistent-return": Array [
    "off",
  ],
  "constructor-super": Array [
    "error",
  ],
  "curly": Array [
    0,
  ],
  "dot-location": Array [
    "off",
  ],
  "dot-notation": Array [
    "warn",
  ],
  "eol-last": Array [
    "off",
  ],
  "eqeqeq": Array [
    "error",
    "smart",
  ],
  "flowtype/boolean-style": Array [
    "off",
  ],
  "flowtype/delimiter-dangle": Array [
    "off",
  ],
  "flowtype/generic-spacing": Array [
    "off",
  ],
  "flowtype/object-type-delimiter": Array [
    "off",
  ],
  "flowtype/semi": Array [
    "off",
  ],
  "flowtype/space-after-type-colon": Array [
    "off",
  ],
  "flowtype/space-before-generic-bracket": Array [
    "off",
  ],
  "flowtype/space-before-type-colon": Array [
    "off",
  ],
  "flowtype/union-intersection-spacing": Array [
    "off",
  ],
  "for-direction": Array [
    "error",
  ],
  "func-call-spacing": Array [
    "off",
  ],
  "function-call-argument-newline": Array [
    "off",
  ],
  "function-paren-newline": Array [
    "off",
  ],
  "generator-star": Array [
    "off",
  ],
  "generator-star-spacing": Array [
    "off",
  ],
  "getter-return": Array [
    "error",
  ],
  "implicit-arrow-linebreak": Array [
    "off",
  ],
  "import/extensions": Array [
    "off",
  ],
  "import/first": Array [
    "warn",
  ],
  "import/no-extraneous-dependencies": Array [
    "off",
  ],
  "import/no-unresolved": Array [
    "off",
  ],
  "import/prefer-default-export": Array [
    "off",
  ],
  "indent": Array [
    "off",
  ],
  "indent-legacy": Array [
    "off",
  ],
  "jsx-a11y/alt-text": Array [
    "off",
  ],
  "jsx-a11y/label-has-for": Array [
    "off",
  ],
  "jsx-a11y/no-noninteractive-element-interactions": Array [
    "off",
  ],
  "jsx-a11y/no-static-element-interactions": Array [
    "off",
  ],
  "jsx-no-target-blank": Array [
    "off",
  ],
  "jsx-quotes": Array [
    "off",
  ],
  "key-spacing": Array [
    "off",
  ],
  "keyword-spacing": Array [
    "off",
  ],
  "linebreak-style": Array [
    "error",
    "unix",
  ],
  "lines-around-comment": Array [
    0,
  ],
  "max-len": Array [
    "off",
  ],
  "multiline-ternary": Array [
    "off",
  ],
  "new-cap": Array [
    "error",
    Object {
      "capIsNew": false,
      "newIsCap": true,
      "properties": true,
    },
  ],
  "new-parens": Array [
    "off",
  ],
  "newline-per-chained-call": Array [
    "off",
  ],
  "no-alert": Array [
    "off",
  ],
  "no-array-constructor": Array [
    "error",
  ],
  "no-arrow-condition": Array [
    "off",
  ],
  "no-async-promise-executor": Array [
    "error",
  ],
  "no-case-declarations": Array [
    "error",
  ],
  "no-class-assign": Array [
    "error",
  ],
  "no-comma-dangle": Array [
    "off",
  ],
  "no-compare-neg-zero": Array [
    "error",
  ],
  "no-cond-assign": Array [
    "warn",
  ],
  "no-confusing-arrow": Array [
    0,
  ],
  "no-console": Array [
    "error",
  ],
  "no-const-assign": Array [
    "error",
  ],
  "no-constant-condition": Array [
    "error",
  ],
  "no-control-regex": Array [
    "error",
  ],
  "no-debugger": Array [
    "error",
  ],
  "no-delete-var": Array [
    "error",
  ],
  "no-dupe-args": Array [
    "error",
  ],
  "no-dupe-class-members": Array [
    "error",
  ],
  "no-dupe-else-if": Array [
    "error",
  ],
  "no-dupe-keys": Array [
    "error",
  ],
  "no-duplicate-case": Array [
    "error",
  ],
  "no-else-return": Array [
    "warn",
  ],
  "no-empty": Array [
    "error",
  ],
  "no-empty-character-class": Array [
    "error",
  ],
  "no-empty-pattern": Array [
    "error",
  ],
  "no-ex-assign": Array [
    "error",
  ],
  "no-extra-boolean-cast": Array [
    "warn",
  ],
  "no-extra-parens": Array [
    "off",
  ],
  "no-extra-semi": Array [
    "off",
  ],
  "no-fallthrough": Array [
    "error",
  ],
  "no-floating-decimal": Array [
    "off",
  ],
  "no-func-assign": Array [
    "error",
  ],
  "no-global-assign": Array [
    "error",
  ],
  "no-import-assign": Array [
    "error",
  ],
  "no-inner-declarations": Array [
    "error",
  ],
  "no-invalid-regexp": Array [
    "error",
  ],
  "no-irregular-whitespace": Array [
    "error",
  ],
  "no-lonely-if": Array [
    "warn",
  ],
  "no-misleading-character-class": Array [
    "error",
  ],
  "no-mixed-operators": Array [
    "warn",
    Object {
      "allowSamePrecedence": true,
      "groups": Array [
        Array [
          "&",
          "|",
          "^",
          "~",
          "<<",
          ">>",
          ">>>",
        ],
        Array [
          "&&",
          "||",
        ],
      ],
    },
  ],
  "no-mixed-spaces-and-tabs": Array [
    "off",
  ],
  "no-multi-spaces": Array [
    "off",
  ],
  "no-multiple-empty-lines": Array [
    "off",
  ],
  "no-nested-ternary": Array [
    "off",
  ],
  "no-new-symbol": Array [
    "error",
  ],
  "no-obj-calls": Array [
    "error",
  ],
  "no-octal": Array [
    "error",
  ],
  "no-param-reassign": Array [
    "error",
  ],
  "no-plusplus": Array [
    "error",
    Object {
      "allowForLoopAfterthoughts": true,
    },
  ],
  "no-prototype-builtins": Array [
    "error",
  ],
  "no-redeclare": Array [
    "error",
  ],
  "no-regex-spaces": Array [
    "error",
  ],
  "no-reserved-keys": Array [
    "off",
  ],
  "no-self-assign": Array [
    "error",
  ],
  "no-setter-return": Array [
    "error",
  ],
  "no-shadow": Array [
    "off",
  ],
  "no-shadow-restricted-names": Array [
    "error",
  ],
  "no-space-before-semi": Array [
    "off",
  ],
  "no-spaced-func": Array [
    "off",
  ],
  "no-sparse-arrays": Array [
    "error",
  ],
  "no-tabs": Array [
    0,
  ],
  "no-this-before-super": Array [
    "error",
  ],
  "no-trailing-spaces": Array [
    "off",
  ],
  "no-undef": Array [
    "error",
  ],
  "no-underscore-dangle": Array [
    "off",
  ],
  "no-unexpected-multiline": Array [
    0,
  ],
  "no-unreachable": Array [
    "error",
  ],
  "no-unsafe-finally": Array [
    "error",
  ],
  "no-unsafe-negation": Array [
    "error",
  ],
  "no-unused-labels": Array [
    "error",
  ],
  "no-unused-vars": Array [
    "error",
    Object {
      "argsIgnorePattern": "^_",
      "varsIgnorePattern": "^_",
    },
  ],
  "no-use-before-define": Array [
    "off",
  ],
  "no-useless-catch": Array [
    "error",
  ],
  "no-useless-constructor": Array [
    "off",
  ],
  "no-useless-escape": Array [
    "off",
  ],
  "no-whitespace-before-property": Array [
    "off",
  ],
  "no-with": Array [
    "error",
  ],
  "no-wrap-func": Array [
    "off",
  ],
  "nonblock-statement-body-position": Array [
    "off",
  ],
  "object-curly-newline": Array [
    "off",
  ],
  "object-curly-spacing": Array [
    "off",
  ],
  "object-property-newline": Array [
    "off",
  ],
  "object-shorthand": Array [
    "warn",
  ],
  "one-var-declaration-per-line": Array [
    "off",
  ],
  "operator-linebreak": Array [
    "off",
  ],
  "padded-blocks": Array [
    "off",
  ],
  "prefer-const": Array [
    "warn",
  ],
  "prefer-template": Array [
    "warn",
  ],
  "quote-props": Array [
    "warn",
    "as-needed",
  ],
  "quotes": Array [
    "off",
  ],
  "react-hooks/exhaustive-deps": Array [
    "warn",
  ],
  "react-hooks/rules-of-hooks": Array [
    "error",
  ],
  "react/display-name": Array [
    2,
  ],
  "react/forbid-prop-types": Array [
    "off",
  ],
  "react/jsx-child-element-spacing": Array [
    "off",
  ],
  "react/jsx-closing-bracket-location": Array [
    "warn",
  ],
  "react/jsx-closing-tag-location": Array [
    "off",
  ],
  "react/jsx-curly-newline": Array [
    "off",
  ],
  "react/jsx-curly-spacing": Array [
    "off",
  ],
  "react/jsx-equals-spacing": Array [
    "off",
  ],
  "react/jsx-filename-extension": Array [
    "warn",
    Object {
      "extensions": Array [
        ".jsx",
        ".tsx",
      ],
    },
  ],
  "react/jsx-first-prop-new-line": Array [
    "off",
  ],
  "react/jsx-indent": Array [
    "warn",
    2,
  ],
  "react/jsx-indent-props": Array [
    "warn",
    2,
  ],
  "react/jsx-key": Array [
    2,
  ],
  "react/jsx-max-props-per-line": Array [
    "off",
  ],
  "react/jsx-newline": Array [
    "off",
  ],
  "react/jsx-no-bind": Array [
    "warn",
    Object {
      "allowArrowFunctions": true,
      "allowBind": false,
      "allowFunctions": false,
      "ignoreDOMComponents": false,
      "ignoreRefs": false,
    },
  ],
  "react/jsx-no-comment-textnodes": Array [
    2,
  ],
  "react/jsx-no-duplicate-props": Array [
    "error",
  ],
  "react/jsx-no-target-blank": Array [
    "warn",
  ],
  "react/jsx-no-undef": Array [
    2,
  ],
  "react/jsx-one-expression-per-line": Array [
    "off",
  ],
  "react/jsx-props-no-multi-spaces": Array [
    "off",
  ],
  "react/jsx-space-before-closing": Array [
    "off",
  ],
  "react/jsx-tag-spacing": Array [
    "off",
  ],
  "react/jsx-uses-react": Array [
    "off",
  ],
  "react/jsx-uses-vars": Array [
    "error",
  ],
  "react/jsx-wrap-multilines": Array [
    "warn",
  ],
  "react/no-array-index-key": Array [
    "error",
  ],
  "react/no-children-prop": Array [
    2,
  ],
  "react/no-danger-with-children": Array [
    2,
  ],
  "react/no-deprecated": Array [
    2,
  ],
  "react/no-did-update-set-state": Array [
    "error",
  ],
  "react/no-direct-mutation-state": Array [
    2,
  ],
  "react/no-find-dom-node": Array [
    "error",
  ],
  "react/no-is-mounted": Array [
    2,
  ],
  "react/no-multi-comp": Array [
    "off",
  ],
  "react/no-render-return-value": Array [
    2,
  ],
  "react/no-string-refs": Array [
    "error",
  ],
  "react/no-unescaped-entities": Array [
    2,
  ],
  "react/no-unknown-property": Array [
    2,
  ],
  "react/no-unsafe": Array [
    0,
  ],
  "react/no-unused-prop-types": Array [
    "off",
  ],
  "react/prop-types": Array [
    "off",
  ],
  "react/react-in-jsx-scope": Array [
    "off",
  ],
  "react/require-default-props": Array [
    "off",
  ],
  "react/require-render-return": Array [
    2,
  ],
  "require-yield": Array [
    "error",
  ],
  "rest-spread-spacing": Array [
    "off",
  ],
  "semi": Array [
    "error",
    "always",
  ],
  "semi-spacing": Array [
    "off",
  ],
  "semi-style": Array [
    "off",
  ],
  "space-after-function-name": Array [
    "off",
  ],
  "space-after-keywords": Array [
    "off",
  ],
  "space-before-blocks": Array [
    "off",
  ],
  "space-before-function-paren": Array [
    "warn",
    Object {
      "anonymous": "always",
      "asyncArrow": "always",
      "named": "never",
    },
  ],
  "space-before-function-parentheses": Array [
    "off",
  ],
  "space-before-keywords": Array [
    "off",
  ],
  "space-in-brackets": Array [
    "off",
  ],
  "space-in-parens": Array [
    "off",
  ],
  "space-infix-ops": Array [
    "off",
  ],
  "space-return-throw-case": Array [
    "off",
  ],
  "space-unary-ops": Array [
    "off",
  ],
  "space-unary-word-ops": Array [
    "off",
  ],
  "standard/array-bracket-even-spacing": Array [
    "off",
  ],
  "standard/computed-property-even-spacing": Array [
    "off",
  ],
  "standard/object-curly-even-spacing": Array [
    "off",
  ],
  "switch-colon-spacing": Array [
    "off",
  ],
  "template-curly-spacing": Array [
    "off",
  ],
  "template-tag-spacing": Array [
    "off",
  ],
  "unicode-bom": Array [
    "off",
  ],
  "unicorn/empty-brace-spaces": Array [
    "off",
  ],
  "unicorn/no-nested-ternary": Array [
    "off",
  ],
  "unicorn/number-literal-case": Array [
    "off",
  ],
  "use-isnan": Array [
    "error",
  ],
  "use-macros/graphql-tag": Array [
    "error",
  ],
  "use-macros/styled-components": Array [
    "error",
  ],
  "valid-typeof": Array [
    "error",
  ],
  "vue/array-bracket-newline": Array [
    "off",
  ],
  "vue/array-bracket-spacing": Array [
    "off",
  ],
  "vue/arrow-spacing": Array [
    "off",
  ],
  "vue/block-spacing": Array [
    "off",
  ],
  "vue/block-tag-newline": Array [
    "off",
  ],
  "vue/brace-style": Array [
    "off",
  ],
  "vue/comma-dangle": Array [
    "off",
  ],
  "vue/comma-spacing": Array [
    "off",
  ],
  "vue/comma-style": Array [
    "off",
  ],
  "vue/dot-location": Array [
    "off",
  ],
  "vue/func-call-spacing": Array [
    "off",
  ],
  "vue/html-closing-bracket-newline": Array [
    "off",
  ],
  "vue/html-closing-bracket-spacing": Array [
    "off",
  ],
  "vue/html-end-tags": Array [
    "off",
  ],
  "vue/html-indent": Array [
    "off",
  ],
  "vue/html-quotes": Array [
    "off",
  ],
  "vue/html-self-closing": Array [
    0,
  ],
  "vue/key-spacing": Array [
    "off",
  ],
  "vue/keyword-spacing": Array [
    "off",
  ],
  "vue/max-attributes-per-line": Array [
    "off",
  ],
  "vue/max-len": Array [
    0,
  ],
  "vue/multiline-html-element-content-newline": Array [
    "off",
  ],
  "vue/mustache-interpolation-spacing": Array [
    "off",
  ],
  "vue/no-extra-parens": Array [
    "off",
  ],
  "vue/no-multi-spaces": Array [
    "off",
  ],
  "vue/no-spaces-around-equal-signs-in-attribute": Array [
    "off",
  ],
  "vue/object-curly-newline": Array [
    "off",
  ],
  "vue/object-curly-spacing": Array [
    "off",
  ],
  "vue/object-property-newline": Array [
    "off",
  ],
  "vue/operator-linebreak": Array [
    "off",
  ],
  "vue/script-indent": Array [
    "off",
  ],
  "vue/singleline-html-element-content-newline": Array [
    "off",
  ],
  "vue/space-in-parens": Array [
    "off",
  ],
  "vue/space-infix-ops": Array [
    "off",
  ],
  "vue/space-unary-ops": Array [
    "off",
  ],
  "vue/template-curly-spacing": Array [
    "off",
  ],
  "wrap-iife": Array [
    "off",
  ],
  "wrap-regex": Array [
    "off",
  ],
  "yield-star-spacing": Array [
    "off",
  ],
}
`;

exports[`eslint-config-wantedly should match snapshot for: settings 1`] = `Object {}`;
